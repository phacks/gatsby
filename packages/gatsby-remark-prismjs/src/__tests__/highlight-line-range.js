const highlightLineRange = require(`../highlight-line-range`)
const fixtures = require(`./fixtures`)
const output = highlighted => highlighted.map(({ code }) => code).join(`\n`)
const getHighlighted = lines => lines.filter(line => line.highlighted)
describe(`highlighting a line range`, () => {
  describe(`highlight-line`, () => {
    it(`strips directive`, () => {
      const highlights = highlightLineRange(fixtures.highlightLine)
      expect(output(highlights)).not.toContain(`highlight-line`)
    })
    it(`highlights line`, () => {
      const highlights = highlightLineRange(fixtures.highlightLine)
      expect(output(getHighlighted(highlights))).toMatchSnapshot()
    })
  })
  describe(`highlight-next-line`, () => {
    it(`strips directive`, () => {
      const highlights = highlightLineRange(fixtures.highlightNextLine)
      expect(output(highlights)).not.toContain(`highlight-next-line`)
    })
    it(`highlights correct line`, () => {
      const highlights = highlightLineRange(fixtures.highlightNextLine)
      expect(output(getHighlighted(highlights))).toMatchSnapshot()
    })
  })
  describe(`highlight-start / highlight-end`, () => {
    it(`strips directives`, () => {
      const highlights = highlightLineRange(fixtures.highlightStartEnd)
      const code = output(highlights)
      ;[`highlight-start`, `highlight-end`].forEach(directive => {
        expect(code).not.toContain(directive)
      })
    })
    it(`highlights correct lines`, () => {
      const highlights = highlightLineRange(fixtures.highlightStartEnd)
      expect(output(getHighlighted(highlights))).toMatchSnapshot()
    })
    it(`highlights without end directive`, () => {
      const highlights = highlightLineRange(fixtures.highlightStartWithoutEnd)
      expect(output(getHighlighted(highlights))).toMatchSnapshot()
    })
  })
  describe(`jsx comment`, () => {
    it(`removes directive`, () => {
      const highlights = highlightLineRange(fixtures.highlightJsxComment)
      expect(output(highlights)).not.toContain(`highlight-line`)
    })
    it(`highlights comment line`, () => {
      const highlights = highlightLineRange(fixtures.highlightJsxComment)
      expect(output(getHighlighted(highlights))).toMatchSnapshot()
    })
  })
  describe(`yaml`, () => {
    it(`strips directive`, () => {
      const highlights = highlightLineRange(fixtures.highlightYaml)
      expect(highlights).not.toContain(`highlight-line`)
    })
    it(`highlights yaml`, () => {
      const highlights = highlightLineRange(fixtures.highlightYaml)
      expect(output(getHighlighted(highlights))).toMatchSnapshot()
    })
  })
  describe(`kitchen sink`, () => {
    it.skip(`strips directives`, () => {
      const highlights = highlightLineRange(fixtures.highlightKitchenSink)
      console.log(output(highlights))
    })
  })
})
