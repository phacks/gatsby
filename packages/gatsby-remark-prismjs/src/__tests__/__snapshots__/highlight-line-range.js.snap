// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlighting a line range highlight-line highlights line 1`] = `
"<span class=\\"gatsby-highlight-code-line\\">  return \\"hello world\\"
</span>"
`;

exports[`highlighting a line range highlight-next-line highlights correct line 1`] = `
"<span class=\\"gatsby-highlight-code-line\\">  return \\"hello world\\"
</span>"
`;

exports[`highlighting a line range highlight-range does not highlight and warns if range is invalid 1`] = `""`;

exports[`highlighting a line range highlight-range highlights correct lines 1`] = `
"<span class=\\"gatsby-highlight-code-line\\">  a = 1 + 1
</span>
<span class=\\"gatsby-highlight-code-line\\">  b = 2 + 2
</span>
<span class=\\"gatsby-highlight-code-line\\">  return \\"hello world\\"
</span>"
`;

exports[`highlighting a line range highlight-range highlights until end of code block if ranges goes farther 1`] = `
"<span class=\\"gatsby-highlight-code-line\\">  a = 1 + 1
</span>
<span class=\\"gatsby-highlight-code-line\\">  b = 2 + 2
</span>
<span class=\\"gatsby-highlight-code-line\\">  return \\"hello world\\"
</span>
<span class=\\"gatsby-highlight-code-line\\">}
</span>
<span class=\\"gatsby-highlight-code-line\\">
</span>"
`;

exports[`highlighting a line range highlight-start / highlight-end highlights correct lines 1`] = `
"<span class=\\"gatsby-highlight-code-line\\">  var a = \\"b\\"
</span>
<span class=\\"gatsby-highlight-code-line\\">  return a + \\"hello world\\"
</span>"
`;

exports[`highlighting a line range highlight-start / highlight-end highlights without end directive 1`] = `
"<span class=\\"gatsby-highlight-code-line\\">var a = \\"b\\"
</span>
<span class=\\"gatsby-highlight-code-line\\">return a + \\"hello world\\"
</span>
<span class=\\"gatsby-highlight-code-line\\">
</span>"
`;

exports[`highlighting a line range jsx comment highlights comment line 1`] = `
"<span class=\\"gatsby-highlight-code-line\\">      <button>sup</button>
</span>"
`;

exports[`highlighting a line range yaml highlights yaml 1`] = `
"<span class=\\"gatsby-highlight-code-line\\">- title: catorce
</span>"
`;
